src/components/Utils/queries.js

```diff
@@ -351,12 +357,12 @@ export const downloadDataFromGuppy = (
         'Content-Type': 'application/json',
       },
       body: JSON.stringify(queryBody),
-    }).then((response) => response.json());
+    }).then((res) => (JSON_FORMAT ? res.json() : jsonToFormat(res.json(), format)));
   }
-  return askGuppyForRawData(path, type, fields, filter, sort, 0, totalCount, accessibility)
+  return askGuppyForRawData(path, type, fields, filter, sort, 0, totalCount, accessibility, format)
     .then((res) => {
       if (res && res.data && res.data[type]) {
-        return res.data[type];
+        return JSON_FORMAT ? res.data[type] : jsonToFormat(res.data[type], format);
```

Comment:
I still think the approiate way is to make this happens in the server side, more specifically in the `getData` and `downloadData` functions https://github.com/uc-cdis/guppy/blob/feat/guppy-format-arg/src/server/es/index.js#L393-L444 , rather than in component side.

This would be the similar to the behavior of the current Sheepdog API, which allows you send format as part of query and get approiate response back. For example, curl https://christopher.planx-pla.net/api/v0/submission/DEV/test/export?node_label=participant&format=tsv and https://christopher.planx-pla.net/api/v0/submission/DEV/test/export?node_label=participant&format=json will have different response body format

If you put this logic in the component side then it will only work for Portal, not for API

Reply:
Yeah I agree, let me move the conversion service to the server. I can return JSON regardless and fill the JSON  "data/type"  property with the actual TSV or CSV data.

Reply:
Update, conversion service will remain on the frontend due to graphql limitations. See TECHDEBT.md for details.